migrations/sql/V2019.06.26.10.30__add_code_required_report_tables.sql

```diff
@@ -0,0 +1,154 @@
+    create_user                          character varying(60)                   NOT NULL   ,
+    create_timestamp                     timestamp with time zone DEFAULT now()  NOT NULL   ,
+    update_user                          character varying(60)                   NOT NULL   ,
+    update_timestamp                     timestamp with time zone DEFAULT now()  NOT NULL   ,
+
+    FOREIGN KEY (compliance_article_id) REFERENCES compliance_article(compliance_article_id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_report_due_date_type) REFERENCES mine_report_due_date_type(mine_report_due_date_type) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_definition is 'Definitions of code required reports, due dates and articles of the code that apply to them.';
+ALTER TABLE mine_report_definition OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report
+(
+    mine_report_id               serial                                   NOT NULL PRIMARY KEY,
+    mine_report_guid             uuid                                     NOT NULL            ,
+    mine_report_definition_id    integer                                  NOT NULL            ,
+    mine_guid                    uuid                                     NOT NULL            ,
+    permit_id                    integer                                                      ,
+    due_date                     timestamp                                NOT NULL            ,
+    submission_year              integer                                                      ,
+    deleted_ind                  boolean DEFAULT false                    NOT NULL            ,
+    create_user                  character varying(60)                    NOT NULL            ,
+    create_timestamp             timestamp with time zone DEFAULT now()   NOT NULL            ,
+    update_user                  character varying(60)                    NOT NULL            ,
+    update_timestamp             timestamp with time zone DEFAULT now()   NOT NULL            ,
+    
+    FOREIGN KEY (mine_report_definition_id) REFERENCES mine_report_definition(mine_report_definition_id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_guid) REFERENCES mine(mine_guid) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (permit_id) REFERENCES permit(permit_id) DEFERRABLE INITIALLY DEFERRED
+
+);
+
+COMMENT ON TABLE mine_report is 'A code required report for a specific mine or permit on a mine.';
+ALTER TABLE mine_report OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_submission (
+    mine_report_submission_id             serial                                  NOT NULL PRIMARY KEY,
+    mine_report_submission_guid           uuid                                    NOT NULL            ,
+    mine_report_id                        integer                                 NOT NULL            ,
+    received_date                         timestamp                                                   ,
```

Comment:
We may want to describe what "received date" means, in a column comment. Is it the date the submission was made? Or when the ministry received the submission? Or are those one in the same?

Reply:
The way this was mad is that this is the date the submission was made in the system.